File: back-end/app/Models/Player.php
<?php

namespace App\Models;

use Carbon\Carbon;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Player extends Model
{
    /** @use HasFactory<\Database\Factories\PlayerFactory> */
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'name',
        'jersey_number',
        'position',
        'date_of_birth',
        'team_id',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'date_of_birth' => 'date',
        ];
    }

    /**
     * Get the team that owns this player.
     *
     * @return BelongsTo<Team>
     */
    public function team(): BelongsTo
    {
        return $this->belongsTo(Team::class);
    }

    /**
     * Get the player's age calculated from date_of_birth.
     * Age is never stored in the database — it is always calculated dynamically.
     *
     * @return int|null
     */
    public function getAgeAttribute(): ?int
    {
        if ($this->date_of_birth) {
            return Carbon::parse($this->date_of_birth)->age;
        }

        return null;
    }

    /**
     * Append the age attribute to the model's JSON representation.
     *
     * @var array
     */
    protected $appends = ['age'];
}

File: front-end/lib/models/player.dart
enum PlayerRole {
  gk('GK'),
  lb('LB'),
  cb('CB'),
  rb('RB'),
  dm('DM'),
  cm('CM'),
  am('AM'),
  lw('LW'),
  rw('RW'),
  st('ST');

  const PlayerRole(this.label);

  final String label;
}

class Player {
  const Player({
    required this.id,
    required this.name,
    required this.jerseyNumber,
    required this.position,
  });

  final int id;
  final String name;
  final int jerseyNumber;
  final PlayerRole position;

  String get numberLabel => '#$jerseyNumber';

  Player copyWith({
    int? id,
    String? name,
    int? jerseyNumber,
    PlayerRole? position,
  }) {
    return Player(
      id: id ?? this.id,
      name: name ?? this.name,
      jerseyNumber: jerseyNumber ?? this.jerseyNumber,
      position: position ?? this.position,
    );
  }
}